Sources/MongoSwift/ReadConcern.swift

```diff
@@ -86,13 +115,19 @@ public class ReadConcern: Codable {
         try container.encodeIfPresent(self.level, forKey: .level)
     }
 
-    /// Cleans up internal state.
-    deinit {
-        guard let readConcern = self._readConcern else {
-            return
+    /**
+     * Creates a new `ReadConcern` with the provided options and passes it to the provided closure.
+     * The read concern is only valid within the body of the closure and will be ended after the body completes.
+     */
+    public func withMongocReadConcern<T>(_ body: (OpaquePointer) throws -> T) throws -> T {
+        var readConcern: OpaquePointer = mongoc_read_concern_new()
```

Comment:
1. does this need to be mutable or can we declare it with `let`? (while we do mutate the value that it points to, I don't think we mutate the pointer itself.)
2. I think you can omit the `: OpaquePointer` type annotation. The compiler should infer it based on the return type of `mongoc_read_concern_new`.